src/windows.rs

```diff
@@ -86,238 +121,318 @@ impl crate::Brightness for Brightness {
-                    if display_devices.len() != physical_monitors.len() {
-                        // There doesn't seem to be any way to directly associate a physical monitor
-                        // handle with the equivalent display device, other than by array indexing
-                        // https://stackoverflow.com/questions/63095216/how-to-associate-physical-monitor-with-monitor-deviceid
-                        return vec![Err(SysError::EnumerationMismatch)];
-                    }
-
-                    physical_monitors
-                        .into_iter()
-                        .zip(display_devices)
-                        .filter_map(|((physical_monitor, description), mut display_device)| {
-                            // Get a file handle for this physical monitor
-                            // Note that this is a different type of handle
-                            let device_name = wchar_to_string(&display_device.DeviceName);
-                            let handle = CreateFileW(
-                                PWSTR(display_device.DeviceID.as_mut_ptr()),
-                                FILE_ACCESS_FLAGS(GENERIC_READ | GENERIC_WRITE),
-                                FILE_SHARE_READ | FILE_SHARE_WRITE,
-                                ptr::null_mut(),
-                                OPEN_EXISTING,
-                                FILE_FLAGS_AND_ATTRIBUTES(0),
-                                HANDLE::NULL,
-                            );
-                            if handle.is_invalid() {
-                                let e = HRESULT::from_thread();
-                                // This error occurs for virtual devices e.g. Remote Desktop
-                                // sessions - they are not real monitors
-                                if e == HRESULT::from_win32(ERROR_ACCESS_DENIED.0) {
-                                    return None;
-                                }
-                                return Some(Err(
-                                    SysError::OpeningMonitorDeviceInterfaceHandleFailed {
-                                        device_name,
-                                        source: e.into(),
-                                    },
-                                ));
-                            }
-                            Some(Ok(Brightness {
-                                physical_monitor,
-                                file_handle: WrappedFileHandle(handle),
-                                device_name,
-                                device_description: wchar_to_string(&description),
-                                device_key: wchar_to_string(&display_device.DeviceKey),
-                            }))
-                        })
-                        .collect()
-                }))
-                .right_stream()
-            }
+        .map_err(|e| SysError::GetMonitorInfoFailed(e))?;
+    Ok((0..)
+        .map(|device_number| {
```

Comment:
I think this was your code lol, anyway it is now fixed.

Reply:
`map_while` was added in the latest Rust 1.57.0, so I think it didn't exist when this code was originally written. Since this code appears in the diff as modified (although it might have just been moved), I reviewed it and thought it would be a nice simplification.